internal/pkg/service/objectstorage/cloud/aws/iam.go

```diff
@@ -14,22 +14,23 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/giantswarm/object-storage-operator/api/v1alpha1"
-	"github.com/giantswarm/object-storage-operator/internal/pkg/managementcluster"
 )
 
 type IAMAccessRoleServiceAdapter struct {
-	iamClient         *iam.Client
-	logger            logr.Logger
-	accountId         string
-	managementCluster managementcluster.ManagementCluster
+	iamClient   *iam.Client
+	logger      logr.Logger
+	accountId   string
+	baseDomain  string
+	clusterName string
 }
 
-func NewIamService(iamClient *iam.Client, logger logr.Logger, accountId string, managementCluster managementcluster.ManagementCluster) IAMAccessRoleServiceAdapter {
+func NewIamService(iamClient *iam.Client, logger logr.Logger, accountId string, baseDomain string, clusterName string) IAMAccessRoleServiceAdapter {
```

Comment:
maybe pass the managemnet cluster and extract the name and base domain from it ? Ideally, we try to not have more than 4 arguments to functions